test/unittests/stack_test.cpp

```diff
@@ -48,3 +52,95 @@ TEST(stack, drop_and_peek)
     EXPECT_EQ(stack.size(), 0);
     EXPECT_TRUE(stack.empty());
 }
+
+TEST(stack, clear)
+{
+    Stack<char> stack;
+    stack.push('a');
+    stack.push('b');
+
+    EXPECT_FALSE(stack.empty());
+    EXPECT_EQ(stack.size(), 2);
+
+    stack.drop(0);
+    EXPECT_FALSE(stack.empty());
+    EXPECT_EQ(stack.size(), 2);
+
+    stack.clear();
+    EXPECT_TRUE(stack.empty());
+    EXPECT_EQ(stack.size(), 0);
+}
+
+TEST(stack, resize)
+{
+    Stack<char> stack;
+    stack.push('a');
+    stack.push('b');
+
+    EXPECT_FALSE(stack.empty());
+    EXPECT_EQ(stack.size(), 2);
+
+    // grow stack
+    stack.resize(4);
+    EXPECT_FALSE(stack.empty());
+    EXPECT_EQ(stack.peek(), 0);
+    EXPECT_EQ(stack.peek(1), 0);
+    EXPECT_EQ(stack.size(), 4);
+
+    stack.drop();
+    EXPECT_FALSE(stack.empty());
+    EXPECT_EQ(stack.size(), 3);
+
+    // shrink stack
+    stack.resize(1);
+    EXPECT_FALSE(stack.empty());
+    EXPECT_EQ(stack.size(), 1);
+
+    stack.clear();
+    EXPECT_TRUE(stack.empty());
+}
+
+TEST(stack, iterator)
+{
+    Stack<char> stack;
+    stack.push('a');
+    stack.push('b');
+    stack.push('c');
+
+    EXPECT_FALSE(stack.empty());
+    EXPECT_EQ(stack.size(), 3);
+
+    auto bottom_item = stack.begin();
+    EXPECT_EQ(*bottom_item, 'a');
+
+    auto top_item = stack.end() - 1;
+    EXPECT_EQ(*top_item, 'c');
+}
+
+// This crashes.
+TEST(stack, DISABLED_peek_on_empty)
+{
+    Stack<char> stack;
+    stack.peek();
+}
+
+TEST(stack, clear_on_empty)
+{
+    Stack<char> stack;
+    stack.clear();
+}
+
+// This crashes.
+TEST(stack, DISABLED_drop_on_empty)
+{
+    Stack<char> stack;
+    stack.drop();
```

Comment:
This should never pass.

Reply:
This uses `resize` internally, so it actually should just work, except if `size() - num_elements` is negative.